core/util.go

```diff
@@ -422,3 +423,14 @@ func RetryBackoff(retries int, base, max time.Duration, factor float64) time.Dur
 	backoff *= (1 - retryJitter) + 2*retryJitter*mrand.Float64()
 	return time.Duration(backoff)
 }
+
+// HashNames returns the sha256 hash of the lowercased, comma joined
+// slice of DNS names
+func HashNames(names []string) []byte {
+	sort.Strings(names)
```

Comment:
You'll want to lower the strings before sorting.

Else, you'll get mismatches because of capitalization causing mis-sorts

Also, add tests.